src/main/java/org/jabref/model/search/rules/SearchRule.java

```diff
@@ -1,13 +1,21 @@
 package org.jabref.model.search.rules;
 
+import java.text.Normalizer;
+
 import org.jabref.model.entry.BibEntry;
 import org.jabref.model.pdf.search.PdfSearchResults;
 
-public interface SearchRule {
+public abstract class SearchRule {
+
+    public abstract boolean applyRule(String query, BibEntry bibEntry);
 
-    boolean applyRule(String query, BibEntry bibEntry);
+    public abstract PdfSearchResults getFulltextResults(String query, BibEntry bibEntry);
 
-    PdfSearchResults getFulltextResults(String query, BibEntry bibEntry);
+    public abstract boolean validateSearchStrings(String query);
 
-    boolean validateSearchStrings(String query);
+    // convert the given string to ASCII string
+    static String convertToASCII(String string) {
+        String normalizedString = Normalizer.normalize(string, Normalizer.Form.NFD);
+        return normalizedString.replaceAll("[^\\p{ASCII}]", "");
+    }
```

Comment:
Similar to @Siedlerchr's comment: Move such helper method to `StringUtils`.

There is IMHO need to convert `SearchRule` from an interface to a class.

Reply:
Will do!